streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -155,9 +155,6 @@ void handleCorruption(final Map<TaskId, Collection<TopicPartition>> taskWithChan
             final TaskId taskId = entry.getKey();
             final Task task = tasks.get(taskId);
 
-            // this call is idempotent so even if the task is only CREATED we can still call it
-            changelogReader.remove(task.changelogPartitions());
```

Comment:
I lost track of how this is now handled. Can you enlighten me?

Reply:
Yeah, so `remove` is basically the opposite of `register` in the ChangelogReader and previously the registration was done by the ProcessorStateManager  while the TaskManager  was responsible for unregistering changelogs (via `remove`).
I renamed `remove` to `unregister` and moved the unregistration to the ProcessorStateManager as well, so the changelog lifecycle is entirely managed by the state manager now.